lxd/db/generate/db/stmt.go

```diff
@@ -289,8 +289,18 @@ func (s *Stmt) create(buf *file.Buffer, replace bool) error {
 
 	for i, field := range fields {
 		if field.IsScalar() {
-			ref := lex.Snake(field.Name)
-			columns[i] = ref + "_id"
+			joinTable, _, err := field.ScalarTableColumn()
+			if err != nil {
+				return err
+			}
+
+			ref := lex.Singular(joinTable)
+			referenceColumn := field.Config.Get("joinon")
+			if referenceColumn == "" {
+				referenceColumn = ref + "_id"
+			}
+
+			columns[i] = referenceColumn
 			table := entityTable(ref, "")
```

Comment:
Can you please remove this `table := entityTable(ref, "")` and just use `joinTable` above? Since we're specifying the table name anyway, we shouldn't try to generate it.

Reply:
I'll push a commit to @markylaing that does that so we don't have to wait until tomorrow